### ISSUE TITLE
bun run priority issue

### ISSUE THREAD

**Issue by eslym** [oven-sh/bun#4888]
### What version of Bun is running?

1.0.0+822a00c4d508b54f650933a73ca5f4a3af9a7983

### What platform is your computer?

Linux 5.15.90.1-microsoft-standard-WSL2 x86_64 unknown

### What steps can reproduce the bug?

execute `bun --bun run build` when there is a directory `./build` and a script `build` presenting at the same time

### What is the expected behavior?

bun should execute the script `build` rather than the directory `./build`, should only run the `./build` directory when explicitly specify with `bun --bun run ./build`

### What do you see instead?

bun is executing the directory `./build`

```shell
vscode ➜ /workspaces/debian/test $ bun --bun run build
Listening on 0.0.0.0:3000
```

### Additional information

there is no such problem without the `--bun` options specified as it use nodejs as runtime
```shell
vscode ➜ /workspaces/debian/test $ bun run build
$ vite build
```

**Comment by eslym** [comment]
i don't know who fixed it, but it no longer present in v1.0.2

**Comment by AlexAllocated** [comment]
I'm encountering this exact issue. v1.0.13
